bin/node_dump

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+set -euo pipefail
+shopt -s nullglob
+
+ROOT_DIR="$(cd "$(dirname "$(readlink "$0" || echo "$0")")"/..; pwd -P)"
+
+echo "Running node dump in ${ROOT_DIR}"
+
+cd "${ROOT_DIR}"
+
+DUMP="log/node_dump_$(date +"%y%m%d_%H%M%S").tar.gz"
+
+collect() {
+    echo "========================================================"
+    echo "    $*"
+    echo "========================================================"
+    eval "$*" || echo "Unavailable"
+    echo -e '\n'
+}
+
+{
+    collect bin/emqx_ctl broker
+    collect bin/emqx eval "'emqx_node_dump:sys_info()'"
+
+    collect uname -a
+    collect uptime
+    collect free -h
+    collect netstat -tnl
+
+    collect bin/emqx_ctl plugins list
+    collect bin/emqx_ctl modules list
+
+    collect bin/emqx_ctl vm all
+    collect bin/emqx_ctl listeners
+} > log/sysinfo.txt
+
+bin/emqx eval 'emqx_node_dump:app_env_dump()' > log/conf.dump
+
+tar czf "${DUMP}" log/*.log.* log/run_erl.log* log/sysinfo.txt log/conf.dump
```

Comment:
logs can be quite large.
maybe as two files, one for sysinfo.txt + conf.dump, another for logs.

Reply:
The goal is to reduce friction, so I so having a single file is beneficial. I can modify the script to avoid collecting log files that are too old.